website/docs/components/code-editor/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,104 @@
+## Usage
+
+### When to use
+
+- When a user needs to create code from scratch.
+- When code requires modification from an existing source.
+
+### When not to use
+
+- As a read-only reference for code, use the [Code Block](/components/code-block) instead.
+
+## Standalone
+
+The `isStandalone` property increases the portability of the Code Editor to ensure that it can be used in different contexts. For example, a common use case of the Code Editor is in a "standalone" context, which can be part of a form, multi-step process, and is generally a part of the normal layout flow.
+
+![Code editor with rounded corners for enabled standalone property](/assets/components/code-editor/code-editor-rounded-standalone.png)
+
+Sometimes it may be necessary to use the Code Editor in a more dense layout or nested within another component.
+
+![Code editor embedded inside of a larger UI, where it is highlighted where the user is actively editing.](/assets/components/code-editor/code-editor-block-level.png)
+
+## Header
+
+The header is optionally available within the Code Editor. It consists of three sections that can be toggled as needed: the text content that includes the title and description, an actions container for secondary actions, and a custom yielded section of primary actions.
```

Comment:
[nit] You _could_ format these as bullets if you wanted since it's three options, but not opinionated.
```suggestion
The header is optionally available within the Code Editor. It consists of three sections that can be toggled as needed:

- the text content that includes the title and description
- an actions container for secondary actions
- a custom yielded section of primary actions.
```